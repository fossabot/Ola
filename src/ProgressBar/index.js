import React from 'react'
import {default as PT} from 'prop-types'
import cx from 'classnames'
import { getElementType } from '../utils'

const ProgressBar = ({ extraClass, children, descriptionPosition, ...props }) => {
  const ElementType = getElementType(ProgressBar, {...props})
  return (
    <div className={cx('ola_progressBar', `is-description-${descriptionPosition}`, extraClass)}>
      { children && <p className="ola_progressBar-description">{children}</p> }
      <ElementType {...props} />
    </div>
  )
}

ProgressBar.defaultProps = {
  as: 'progress',
  extraClass: null,
  children: null,
  descriptionPosition: 'top'
}

ProgressBar.propTypes = {
  as: PT.oneOf(['progress', 'meter']),
  /** Extra className */
  extraClass: PT.string,
  /** Description Position */
  descriptionPosition: PT.oneOf(['top', 'right']),
  /** Childen nodes */
  children: PT.oneOfType([
    PT.string,
    PT.arrayOf(PT.node),
    PT.node
  ])
}

export default ProgressBar
