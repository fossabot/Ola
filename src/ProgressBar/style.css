.ola_progressBar {
  --color: var(--brand);
  --background: var(--white);

  width: 100%;
  display: flex;
  flex-wrap: wrap;
  flex-direction: row-reverse;

  &.is-description-right {
    align-items: baseline;
    flex-wrap: nowrap;
    white-space: nowrap;
  }
}

.ola_progressBar-description {
  font: var(--font-caption);
  color: var(--gray);
  margin: 0px 0px var(--size-3) var(--size-3);
}

.ola_progressBar-description strong {
  color: var(--black);
  font-weight: var(--bold);
}

.ola_progressBar progress {
    appearance: none;
    border: solid 1px var(--color);
    height: 10px;
    width: 100%;
    min-width: 100px;
    background: var(--background);
    border-radius: 6px;
    box-sizing: border-box;

    &::-webkit-progress-value {
        background: var(--color);
    }
    &::-webkit-progress-bar {
        background: transparent;
    }
    &::-moz-progress-bar {
        background: var(--color);
    }

    &:indeterminate {
        --color: var(--gray-xlight);
        --background: var(--gray-xlight);
    }
}

.ola_progressBar meter {
    -moz-appearance: none;
    border-radius: 6px;
    box-sizing: border-box;
    background: var(--background);
    width: 100%;
    height: 10px;

    &::-moz-meter-bar {
        background: none;
        background-color: var(--color);
    }
    &::-moz-meter-optimum::-moz-meter-bar {
        background-color: var(--color);
    }
    &::-moz-meter-sub-optimum::-moz-meter-bar {
        background-color: var(--warning);
    }
    &::-moz-meter-sub-sub-optimum::-moz-meter-bar {
        background-color: var(--error);
    }

    &::-webkit-meter-bar {
        height: 10px;
        border-radius: 6px;
        background: none;
        border: solid 1px var(--color);
    }
    &::-webkit-meter-optimum-value {
        background: var(--color);
    }
    &::-webkit-meter-suboptimum-value {
        background: var(--warning);
    }
    &::-webkit-meter-even-less-good-value {
        background: var(--error);
    }
    &:not(value) {
        --color: var(--gray-xlight);
        --background: var(--gray-xlight);
    }
}
/* Only firefox */
@-moz-document url-prefix() {
    .ola_progressBar meter {
        border: solid 1px var(--color);
    }
}
