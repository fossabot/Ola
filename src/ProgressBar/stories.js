import React from 'react'
import { storiesOf } from '@storybook/react'

import ProgressBar from './'

storiesOf('ProgressBar')
  .add('Progress element', () => (
    <div>
      <figure><ProgressBar value="20" max="100" /></figure>
      <figure><ProgressBar value="50" max="100">Loading...</ProgressBar></figure>
      <figure>
        <ProgressBar value="80" max="100" descriptionPosition="right">
          <strong>20%</strong> completed
        </ProgressBar>
      </figure>
    </div>
  ))
  .add('Meter element', () => (
    <div>
      <figure><ProgressBar as="meter" value="120" min="100" max="200" low="130" /></figure>
      <figure><ProgressBar as="meter" value="150" min="100" max="200">150 / 200</ProgressBar></figure>
      <figure>
        <ProgressBar as="meter" value="180" min="100" max="200" descriptionPosition="right">
          <strong>180 / 200</strong>
        </ProgressBar>
      </figure>
    </div>
  ))