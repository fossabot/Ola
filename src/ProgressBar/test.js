import React from 'react'
import ProgressBar from './'
import renderer from 'react-test-renderer'


it('Default ProgressBar', () => {
  const tree = renderer
    .create(
      <ProgressBar value="100" />
    )
    .toJSON()

  expect(tree).toMatchSnapshot()
})

it('ProgressBar with description', () => {
  const tree = renderer
    .create(
      <ProgressBar value="100">This is a <strong>description</strong></ProgressBar>
    )
    .toJSON()

  expect(tree).toMatchSnapshot()
})
