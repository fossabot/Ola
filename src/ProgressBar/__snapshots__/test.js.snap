// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default ProgressBar 1`] = `
<div
  className="ola_progressBar is-description-top"
>
  <progress
    as="progress"
    value="100"
  />
</div>
`;

exports[`ProgressBar with description 1`] = `
<div
  className="ola_progressBar is-description-top"
>
  <p
    className="ola_progressBar-description"
  >
    This is a 
    <strong>
      description
    </strong>
  </p>
  <progress
    as="progress"
    value="100"
  />
</div>
`;
